File: app/src/main/java/com/appzonegroup/app/fasttrack/model/WithdrawalRequest.java
package com.appzonegroup.app.fasttrack.model;

/**
 * Created by Oto-obong on 17/07/2017.
 */

public class WithdrawalRequest {

    private String AgentPhoneNumber;
    private String InstitutionCode;

    private String CustomerAccountNumber;
    private String AgentPin;

    private String Token;
    private String CustomerPin = "0000";
    private String Amount;

    private String GeoLocation;
    private String AdditionalInformation;

    public String getAgentPhoneNumber() {
        return AgentPhoneNumber;
    }

    public void setAgentPhoneNumber(String agentPhoneNumber) {
        AgentPhoneNumber = agentPhoneNumber;
    }

    public String getInstitutionCode() {
        return InstitutionCode;
    }

    public void setInstitutionCode(String institutionCode) {
        InstitutionCode = institutionCode;
    }

    public String getCustomerAccountNumber() {
        return CustomerAccountNumber;
    }

    public void setCustomerAccountNumber(String customerAccountNumber) {
        CustomerAccountNumber = customerAccountNumber;
    }

    public String getAgentPin() {
        return AgentPin;
    }

    public void setAgentPin(String agentPin) {
        AgentPin = agentPin;
    }

    public String getToken() {
        return Token;
    }

    public void setToken(String token) {
        Token = token;
    }

    public String getCustomerPin() {
        return CustomerPin;
    }

    public void setCustomerPin(String customerPin) {
        CustomerPin = customerPin;
    }

    public String getGeoLocation() {
        return GeoLocation;
    }

    public void setGeoLocation(String geoLocation) {
        GeoLocation = geoLocation;
    }

    public String getAdditionalInformation() {
        return AdditionalInformation;
    }

    public void setAdditionalInformation(String additionalInformation) {
        AdditionalInformation = additionalInformation;
    }

    public String getAmount() {
        return Amount;
    }

    public void setAmount(String amount) {
        Amount = amount;
    }
}

File: core/src/main/java/com/creditclub/core/data/request/WithdrawalRequest.kt
package com.creditclub.core.data.request

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class WithdrawalRequest(

    @SerialName("AgentPhoneNumber")
    val agentPhoneNumber: String?,

    @SerialName("InstitutionCode")
    val institutionCode: String?,

    @SerialName("CustomerAccountNumber")
    val customerAccountNumber: String,

    @SerialName("AgentPin")
    val agentPin: String? = null,

    @SerialName("Token")
    val token: String? = null,

    @SerialName("CustomerPin")
    val customerPin: String? = null,

    @SerialName("Amount")
    val amount: String,

    @SerialName("GeoLocation")
    val geoLocation: String? = null,

    @SerialName("AdditionalInformation")
    val additionalInformation: String? = null,

    @SerialName("RetrievalReferenceNumber")
    val retrievalReferenceNumber: String,

    @SerialName("DeviceNumber")
    val deviceNumber: Int = 0,
) {
    @Serializable
    data class Additional(
        @SerialName("Currency")
        val currency: String? = "",

        @SerialName("CustomerPhoneNumber")
        val customerPhoneNumber: String? = null,
    )
}